File: FlexPort Android/app/src/main/java/com/flexport/game/networking/NetworkConfiguration.kt
package com.flexport.game.networking

object NetworkConfiguration {
    // Base URLs - configurable for development vs production
    // Using the same server as web for cross-platform multiplayer
    const val BASE_URL = "https://flexport-multiplayer.herokuapp.com/api/v1"
    const val WEBSOCKET_URL = "wss://flexport-multiplayer.herokuapp.com/ws"
    
    // Timeout configurations (in milliseconds)
    const val REQUEST_TIMEOUT = 30_000L
    const val WEBSOCKET_TIMEOUT = 30_000L
    const val RETRY_DELAY = 2_000L
    
    // Retry configurations
    const val MAX_RETRIES = 3
    
    // API endpoints
    object Endpoints {
        const val MATCHMAKING = "matchmaking"
        const val SYNC_GAME_STATE = "game/sync"
        const val LEADERBOARD = "leaderboard"
        
        fun gameSession(sessionId: String) = "game/session/$sessionId"
        fun playerStats(playerId: String) = "player/$playerId/stats"
    }
    
    // WebSocket ping interval (in milliseconds)
    const val PING_INTERVAL = 30_000L
    
    // Development/debug settings
    const val ENABLE_NETWORK_LOGS = true
    const val ENABLE_WEBSOCKET_LOGS = true
}

File: FlexPort iOS/Sources/FlexPort/Networking/NetworkConfiguration.swift
import Foundation

/// Network configuration for FlexPort game services
struct NetworkConfiguration {
    static let shared = NetworkConfiguration()
    
    // Backend URLs
    let baseURL: URL
    let webSocketURL: URL
    
    // API endpoints
    let apiVersion = "v1"
    
    // Timeouts
    let requestTimeout: TimeInterval = 30.0
    let webSocketTimeout: TimeInterval = 60.0
    
    // Network settings
    let maxRetries = 3
    let retryDelay: TimeInterval = 1.0
    
    private init() {
        // Production URLs - replace with actual backend URLs
        self.baseURL = URL(string: "https://api.flexport-game.com/\(apiVersion)")!
        self.webSocketURL = URL(string: "wss://ws.flexport-game.com/\(apiVersion)")!
    }
    
    // API Endpoints
    enum Endpoint {
        case matchmaking
        case leaderboard
        case gameSession(String)
        case playerStats(String)
        case syncGameState
        
        var path: String {
            switch self {
            case .matchmaking:
                return "/matchmaking"
            case .leaderboard:
                return "/leaderboard"
            case .gameSession(let sessionId):
                return "/sessions/\(sessionId)"
            case .playerStats(let playerId):
                return "/players/\(playerId)/stats"
            case .syncGameState:
                return "/sync"
            }
        }
    }
}

/// Network error types
enum NetworkError: LocalizedError {
    case noConnection
    case timeout
    case serverError(Int)
    case invalidResponse
    case authenticationFailed
    case rateLimited
    case custom(String)
    
    var errorDescription: String? {
        switch self {
        case .noConnection:
            return "No internet connection available"
        case .timeout:
            return "Request timed out"
        case .serverError(let code):
            return "Server error: \(code)"
        case .invalidResponse:
            return "Invalid response from server"
        case .authenticationFailed:
            return "Authentication failed"
        case .rateLimited:
            return "Too many requests. Please try again later"
        case .custom(let message):
            return message
        }
    }
}